### PR TITLE
Add factory and new-style config for Prometheus receiver

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This is part of remaining migration to new configuration format.

Github issue: https://github.com/open-telemetry/opentelemetry-service/issues/33

Testing done: make

### CODE REVIEW

File: receiver/prometheusreceiver/factory.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	return &ConfigV2{
+		ReceiverSettings: configmodels.ReceiverSettings{
+			TypeVal:  typeStr,
+			NameVal:  typeStr,
+			Endpoint: "127.0.0.1:9090",
+		},
+	}
+}
+
+// CreateTraceReceiver creates a  trace receiver based on provided config.
```
Comment (pjanotti) [inline receiver/prometheusreceiver/factory.go]: nit of nitness: double space before `trace` :)

Comment (tigrannajaryan) [reply receiver/prometheusreceiver/factory.go]: Sharp eye! :-)

**Comment by songy23** [comment]
I assume the build failure is unrelated to this change?

**Comment by tigrannajaryan** [comment]
No, build failure is unrelated. We have a bug to fix the build failure: https://github.com/open-telemetry/opentelemetry-service/issues/43

**Pull request merged**